OrbitGl/TrackAccessibility.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TrackAccessibility.h"
+
+#include "GlCanvas.h"
+#include "TimeGraph.h"
+#include "Track.h"
+
+namespace orbit_gl {
+
+using orbit_accessibility::AccessibleInterface;
+
+const AccessibleInterface* AccessibleTrackContent::AccessibleParent() const {
+  return track_->AccessibilityInterface();
+}
+
+std::string AccessibleTrackContent::AccessibleName() const {
+  CHECK(track_ != nullptr);
+  return track_->GetName() + "_content";
+}
+
+AccessibilityRect AccessibleTrackContent::AccessibleLocalRect() const {
+  CHECK(track_ != nullptr);
+  CHECK(track_->GetTimeGraph() != nullptr);
+
+  auto& layout = track_->GetTimeGraph()->GetLayout();
+  return AccessibilityRect(0, static_cast<int>(layout.GetTrackTabHeight()),
+                           static_cast<int>(track_->GetSize()[0]),
+                           static_cast<int>(track_->GetSize()[1]));
+}
+
+AccessibilityState AccessibleTrackContent::AccessibleState() const {
+  AccessibilityState result;
+  return result;
+}
+
+const AccessibleInterface* AccessibleTrackTab::AccessibleParent() const {
+  return track_->AccessibilityInterface();
+}
+
+std::string AccessibleTrackTab::AccessibleName() const {
+  CHECK(track_ != nullptr);
+  return track_->GetName() + "_tab";
+}
+
+AccessibilityRect AccessibleTrackTab::AccessibleLocalRect() const {
+  CHECK(track_ != nullptr);
+  CHECK(track_->GetTimeGraph() != nullptr);
+
+  auto& layout = track_->GetTimeGraph()->GetLayout();
+  return AccessibilityRect(static_cast<int>(layout.GetTrackTabOffset()), 0,
+                           static_cast<int>(layout.GetTrackTabWidth()),
+                           static_cast<int>(layout.GetTrackTabHeight()));
+}
+
+AccessibilityState AccessibleTrackTab::AccessibleState() const {
+  AccessibilityState result;
+  return result;
+}
+
+const AccessibleInterface* AccessibleTrack::AccessibleParent() const {
+  CHECK(track_ != nullptr);
+  return track_->GetTimeGraph()->Accessibility();
+}
+
+std::string AccessibleTrack::AccessibleName() const {
+  CHECK(track_ != nullptr);
+  return track_->GetName();
+}
+
+AccessibilityRect AccessibleTrack::AccessibleLocalRect() const {
+  CHECK(track_ != nullptr);
+  CHECK(track_->GetTimeGraph() != nullptr);
+  CHECK(track_->GetTimeGraph()->GetCanvas() != nullptr);
```

Comment:
I am now instead getting the canvas from the track - this makes more sense in the current state as the tracks depend on properties of the canvas (width, scrolling...), and it can then be removed straightforward later if we either decide to remove this dependency to the canvas, or make `GetCanvas()` part of the generic ui element interface.